server/graphql/v2/object/VirtualCard.ts

```diff
@@ -103,6 +106,50 @@ export const VirtualCard = new GraphQLObjectType({
         }
       },
     },
+    remainingLimit: {
+      type: GraphQLInt,
+      async resolve(virtualCard, _, req) {
+        const collective = await req.loaders.Collective.byId.load(virtualCard.CollectiveId);
+        if (canSeeVirtualCardPrivateInfo(req, collective)) {
+          const { spendingLimitAmount, spendingLimitInterval } = virtualCard;
+          let startOfInterval: string;
+
+          // Stripe spending limit intervals start on UTC midnight for daily, Sunday at midnight UTC for weekly and 1st of the month or year
+          // https://stripe.com/docs/api/issuing/cards/object#issuing_card_object-spending_controls-spending_limits-interval
+          switch (spendingLimitInterval) {
+            case 'DAILY':
+              startOfInterval = moment().utc(true).startOf('day').toISOString();
+              break;
+            case 'WEEKLY':
+              startOfInterval = moment().utc(true).startOf('isoWeek').toISOString();
+              break;
+            case 'MONTHLY':
+              startOfInterval = moment().utc(true).startOf('month').toISOString();
+              break;
+            case 'YEARLY':
+              startOfInterval = moment().utc(true).startOf('year').toISOString();
+              break;
+            case 'ALL_TIME':
+              startOfInterval = undefined;
+              break;
+            case 'PER_AUTHORIZATION':
+              return spendingLimitAmount;
+            default:
+              return null;
+          }
+
+          const sumExpensesInPeriod = await models.Expense.sum('amount', {
+            where: {
+              VirtualCardId: virtualCard.id,
```

Comment:
We should probably add an index on this column to be performant on the production data.

Reply:
Added an index to Expenses